File: javascript/removeVowels.js

/* This function removes all the vowels from a string using recursion
 for detail explanation please watch
 https://youtu.be/QRW43YPDcNM
*/
function removeVowels(str){
    let str = "hello world";
    return removeVowels_helper(str, 0, "");
}

function removeVowels_helper(str, length, result){
  if(str.length==length) return result;
  let vowels = "aeiou";
   if(vowels.includes(str.charAt(length))) 
      return removeVowels_helper(str, length+1, result);
   return removeVowels_helper(str, length+1, result+ str.charAt(length));
}

File: java/src/RemoveVowels.java
public class RemoveVowels {
    public static void main(String[] args) {

        String str1 = "hello world";
        System.out.println(removeVowels(str1,0,""));;
    }
    /*this method removes all the vowels from a given string using recursion*/
    private static String removeVowels(String str, int length, String result){
        if(str.length()==length) return result;
        String vowels = "aeiou";
        if(vowels.contains(String.valueOf(str.charAt(length))))
            return removeVowels(str,length+1, result);
        return removeVowels(str,length+1, result+String.valueOf(str.charAt(length)));
    }
}
